### PR TITLE
Implement feature flag support for snapshots

### PR DESCRIPTION
**Pull request by johnsonw**
Implemenet feature flag support for snapshots

Signed-off-by: johnsonw <wjohnson@whamcloud.com>

---
This change is [Reviewable](https://reviewable.io/reviews/whamcloud/integrated-manager-for-lustre/2278)

### CODE REVIEW

**Comment by johnsonw** [comment]
This shows that snapshots is not available when set to false:

**Comment by johnsonw** [comment]
snapshots_route

**Review by jgrund — DISMISSED** [review]

**Comment by johnsonw** [comment]
I've also tested this in docker as well. Here are screenshots showing that the route is now available when `use_snapshots` is set to true.

image

image

Note that in the second image not only did I try selecting the menu item but I also entered the URL in manually to verify that it would pick up the route correctly. The route does not work and does not display in the config menu if `use_snapshots` is set to false.

**Comment by johnsonw** [comment]
Also on docker when `use_snapshots` is false:
image

image

**Review by jgrund — APPROVED** [review]

**Review by mkpankov — APPROVED** [review]
> Reviewed 7 of 7 files at r1, 1 of 1 files at r3.
> *[Reviewable](https://reviewable.io/reviews/whamcloud/integrated-manager-for-lustre/2278)* status: :shipit: complete! all files reviewed, all discussions resolved (waiting on @johnsonw)

File: iml-gui/crate/src/lib.rs [hunk 1]
```diff
@@ -364,6 +364,10 @@ pub fn update(msg: Msg, model: &mut Model, orders: &mut impl Orders<Msg, GMsg>)
         }
         Msg::LoadPage => {
```
Comment (ip1981) [inline iml-gui/crate/src/lib.rs:365]: Maybe do it earlier on `Msg::RouteChanged`?

File: iml-gui/crate/src/lib.rs [hunk 2]
```diff
@@ -364,6 +364,10 @@ pub fn update(msg: Msg, model: &mut Model, orders: &mut impl Orders<Msg, GMsg>)
...
             if model.loading.loaded() && !model.page.is_active(&model.route) {
+                if model.route == Route::Snapshots && !model.conf.use_snapshots {
+                    model.route = Route::NotFound;
```
Comment (ip1981) [inline iml-gui/crate/src/lib.rs:368]: Maybe do it earlier on Msg::RouteChanged?

Comment (jgrund) [reply iml-gui/crate/src/lib.rs:368]: It won't work because we need to wait for the conf to be loaded.

**Pull request merged**